remove autoconf support

sergiud opened the pull request:
Fixes #537.

drigz commented:
README.md and INSTALL need to be updated as they refer to autoconf. For Bazel, you could add this to README.md (adapted from gflags).

## Bazel

To use glog within a project which uses the [Bazel](https://bazel.build/) build tool, add the following lines to your WORKSPACE file:
```
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "com_github_gflags_gflags",
    sha256 = "34af2f15cf7367513b352bdcd2493ab14ce43692d2dcd9dfc499492966c64dcf",
    strip_prefix = "gflags-2.2.2",
    urls = ["https://github.com/gflags/gflags/archive/v2.2.2.tar.gz"],
)

http_archive(
    name = "com_github_google_glog",
    sha256 = "62efeb57ff70db9ea2129a16d0f908941e355d09d6d83c9f7b18557c0a7ab59e",
    strip_prefix = "glog-d516278b1cd33cd148e8989aec488b6049a4ca0b",
    urls = ["https://github.com/google/glog/archive/d516278b1cd33cd148e8989aec488b6049a4ca0b.zip"],
)
```

You can then add `@com_github_google_glog//:glog` to the deps section of a cc_binary or cc_library rule, and `#include <glog/logging.h>` to include it in your source code. Here's a simple example:

```
cc_binary(
    name = "main",
    srcs = ["main.cc"],
    deps = ["@com_github_google_glog//:glog"],
)
```

drigz commented:
Thanks! The new README is great.

sergiud commented:
I converted the `README` from Markdown to reStructuredText because the latter allows to easily generate TOCs and supports Starlark syntax highlighting used by Bazel. Additionally, I also moved `CONTRIBUTING.md` contents into `README`. The changes should be now final.
